### PR TITLE
Use ASP Development Exception Page

### PR DESCRIPTION
**Pull request by mark-monteiro**
When running the application in development mode, show the built-in ASP development exception page instead of returning a production-friendly result. This page has a lot of useful information for development.

image

**Changes**
* Set `ASPNETCORE_ENVIRONMENT=Development` when running from Visual Studio (this also appears to work for VS Code and the `dotnet run` CLI)
* Do not handle exceptions in `HttpListenerHost` when `ASPNETCORE_ENVIRONMENT=Development` is set so that they bubble up to the appropriate middleware handler
* The configuration to show this page was already set, but the above two changes allow it to actually appear: https://github.com/jellyfin/jellyfin/blob/master/Jellyfin.Server/Startup.cs#L58

### CODE REVIEW

File: Emby.Server.Implementations/HttpServer/HttpListenerHost.cs [hunk 1]
```diff
@@ -530,22 +535,25 @@ public async Task RequestHandler(IHttpRequest httpReq, string urlString, string
...
+                    throw;
+                }
 
-                await ErrorHandler(ex, httpReq, logException).ConfigureAwait(false);
+                bool ignoreStackTrace =
+                    ex is SocketException ||
+                    ex is IOException ||
+                    ex is OperationCanceledException ||
+                    ex is SecurityException ||
+                    ex is FileNotFoundException;
```
Comment (dkanada) [inline Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:555]: Weren't we doing boolean operators at the beginning?

Comment (Bond-009) [reply Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:555]: Yes we are, good catch

File: Emby.Server.Implementations/HttpServer/HttpListenerHost.cs [hunk 2]
```diff
@@ -530,22 +535,25 @@ public async Task RequestHandler(IHttpRequest httpReq, string urlString, string
...
+                    throw;
+                }
 
-                await ErrorHandler(ex, httpReq, logException).ConfigureAwait(false);
+                bool ignoreStackTrace =
+                    ex is SocketException
+                    || ex is IOException
+                    || ex is OperationCanceledException
+                    || ex is SecurityException
+                    || ex is FileNotFoundException;
```
Comment (dkanada) [inline Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:555]: I think you're also supposed to indent these lines but I'm not sure.

Comment (Bond-009) [reply Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:555]: Looks alright like it is now

**Pull request merged**